### PR TITLE
Cleanly close tqdm

### LINKED ISSUE #941: Cleanly close tqdm instance
With `conda create -n ft 'featuretools<0.14.0' tqdm ipykernel`:

```python
#from tqdm.auto import tqdm
from tqdm import tqdm
import pandas as pd
from random import random
import warnings

with warnings.catch_warnings():
    warnings.filterwarnings('ignore', category=FutureWarning)
    tqdm.pandas()

df = pd.DataFrame((random() for _ in range(10)) for _ in range(int(1e6)))

try:
    df.progress_apply(lambda x: x**x ** (1/x) ** x ** (1/x) ** x ** (1/x))
except KeyboardInterrupt:
    print("finished with %d instances left" % len(tqdm._instances))
```
```
60%|██████    | 6/10 [00:02<00:01,  2.77it/s]
finished with 0 instances left
```
```python
#from tqdm.auto import tqdm
from tqdm import tqdm
import featuretools as ft

es = ft.demo.load_mock_customer(return_entityset=True)
cutoff_time = es['transactions'].df[["transaction_id", "transaction_time"]]

try:
    ft.dfs(entityset=es, target_entity='transactions', cutoff_time=cutoff_time, verbose=True)
except KeyboardInterrupt:
    print("finished with %d instances left" % len(tqdm._instances))
```
```
Built 33 features
Elapsed: 00:02 | Progress:   6%|▌         finished with 1 instances left
Elapsed: 00:02 | Progress:   6%|▌         
```

So `tqdm.pandas` works as expected but `featuretools` doesn't.

Looks like `featuretools` may have some error handling. My guess is your custom function which you're handing to `pandas` will itself catch the error and not let it propagate to `tqdm`. If you're doing your own error handling then it's your responsibility to clean up everything, including but not limited to `tqdm`. You'd really need to `close()` the `tqdm` instance rather than hackily clearing the whole internal `_instances` set:

https://github.com/tqdm/tqdm/blob/5e8978909c639bc20f58bf877e66cc6a0b9fb5bf/tqdm/std.py#L766-L767

_Originally posted by @casperdcl in https://github.com/FeatureLabs/featuretools/pull/932#issuecomment-622453513_

### PR DESCRIPTION
**Pull request by frances-h**
- Use `with` to ensure tqdm instance is closed

Resolves #941

### CODE REVIEW

File: featuretools/computational_backends/calculate_feature_matrix.py [hunk 1]
```diff
@@ -283,61 +283,61 @@ def calculate_feature_matrix(features, entityset=None, cutoff_time=None, instanc
 
     progress_bar = make_tqdm_iterator(**tqdm_options)
     progress_bar._instances.clear()
```
Comment (rwedge) [inline featuretools/computational_backends/calculate_feature_matrix.py:285]: do we need this progress bar instance?

Comment (frances-h) [reply featuretools/computational_backends/calculate_feature_matrix.py]: nope, that was accidentally left in with a merge

**Comment by rwedge** [comment]
We also use `make_tqdm_iterator` in `encode_features` -- do we need to use a with statement there as well?

File: featuretools/tests/computational_backend/test_calculate_feature_matrix.py [hunk 2]
```diff
@@ -1637,3 +1645,34 @@ def __call__(self, update, progress_percent, time_elapsed):
...
+                             es,
+                             verbose=True)
+
+    assert len(tqdm._instances) == 0
+
+    try:
+        calculate_feature_matrix([value, error_feature],
+                                 es,
+                                 verbose=True)
+    finally:
```
Comment (rwedge) [inline featuretools/tests/computational_backend/test_calculate_feature_matrix.py:1677]: I think this needs an except clause

**Pull request merged**